src/main/java/codeu/model/store/basic/UserStore.java

```diff
@@ -126,5 +126,30 @@ public boolean isUserRegistered(String username) {
   public void setUsers(List<User> users) {
     this.users = users;
   }
+
+  /**
+   * @return the current number of users.
+   */
+  public int getNumUsers() {
+    return this.users.size();
+  }
+
+
+  public List<String> getTopUsernames() {
+    User oldest = users.get(0);
+    User newest = users.get(0);
+    for (User user : users) {
+      if (user.getCreationTime().compareTo(oldest.getCreationTime()) < 0) {
+        oldest = user;
+      } else if (user.getCreationTime().compareTo(newest.getCreationTime()) > 0) {
+        newest = user;
+      }
+    }
+    List<String> topUsers = new ArrayList<>();
```

Comment:
nit:
topUsers is not the best name for this variable.
Consider:
1) Rename it o oldestAndNewestUsers.
2) Or better, just create to String variable one called oldestUser, the other called newestUser.

Reply:
Or maybe call it edgeUsers, or topAndBottomUsers? But again, I think keep two separate variables would be more readable.